index.bs

```diff
@@ -74,903 +60,149 @@ specification uses that specification and terminology.
-     1. Let |transport| be the {{WebTransport}}, which the |stream| was created
-        from.
-     1. Let |abortInfo| be the first argument.
-     1. Start the closing procedure by sending a message to the remote side
-        indicating that the stream has been aborted (for QUIC, this is a
-        STOP_SENDING frame) with its error code set to the value of
-        |abortInfo.errorCode|.
-
-: <dfn method for="IncomingStream">arrayBuffer()</dfn>
-:: A convenience method that asynchronously reads all the contents of
-   |IncomingStream.readable| and returns it as an ArrayBuffer.  This locks the
-   stream while reading, just as if |IncomingStream.readable.getReader()| were
-   used.
-
-# Interface `WebTransportStream` #  {#web-transport-stream}
+: <dfn attribute for="GenericTransformStream">readable</dfn>
+:: The `readable` attribute's getter, when invoked, must return this object's transform {{[[readable]]}}.
+: <dfn attribute for="GenericTransformStream">writable</dfn>
+:: The `writable` attribute's getter, when invoked, must return this object's transform {{[[writable]]}}.
 
-A collection of common attributes and methods of all streams.
+# Interface `CompressionStream` #  {#compression-stream}
 
 <pre class="idl">
-[ Exposed=Window ]
-interface WebTransportStream {
-    readonly attribute unsigned long long streamId;
-    readonly attribute WebTransport transport;
+[ Exposed=(Window,Worker)]
+interface CompressionStream {
+  constructor(DOMString format);
 };
+CompressionStream includes GenericTransformStream;
 </pre>
 
-## Attributes ##  {#web-transport-stream-attributes}
+The <dfn interface>CompressionStream(format)</dfn> constructor, when invoked, must run these steps:
+    1. If <dfn>*format*</dfn> is unsupported in CompressionStream, then throw a TypeError.
+    1. Let *cs* be a new CompressionStream object.
+    1. Let *transformAlgorithm* be an algorithm which takes a *chunk* argument and runs the <a>compress and enqueue a chunk</a> algorithm with *cs* and *chunk*.
+    1. If *chunk* is not of supported type, then throw a TypeError.
+    1. Let *flushAlgorithm* be an algorithm which takes no argument and runs the <a for=CompressionStream>flush and enqueue</a> algorithm with *cs*.
+    1. Construct TransformStream.
 
-: <dfn for="WebTransportStream" attribute>streamId</dfn>
-:: The readonly attribute referring to the ID of the {{WebTransportStream}}
-   object.
-: <dfn for="WebTransportStream" attribute>transport</dfn>
-:: The readonly attribute referring to the related {{WebTransport}} object.
+The <dfn>compress and enqueue a chunk</dfn> algorithm, given a CompressionStream object *cs* and a *chunk*, runs these steps:
+    1. Let *buffer* be the result of compressing *chunk*. If this throws an exception, then return a promise rejected with that exception.
+    1. Let *controller* be *cs*'s transform.[[TransformStreamController]].
+    1. If *buffer* is non-empty, call {{TransformStreamDefaultController}}(*controller*, *buffer*).
```

Comment:
This needs to be `<a abstract-op>TransformStreamDefaultControllerEnqueue</a>`, not `{{TransformStreamDefaultController}}`.